packages/cli/src/commands/__tests__/update.test.js

```diff
@@ -0,0 +1,342 @@
+//       fs.readFileSync('/tmp/.redwood/update-data.json')
+//     )
+//     expect(updateData).toEqual({
+//       localVersion: '1.0.0',
+//       remoteVersion: '1.0.0',
+//       skipVersion: '0.0.0',
+//       upgradeAvailable: false,
+//       lastChecked: Date.parse('2022-01-02T12:00:00.000Z'),
+//       lastShown: Date.parse('2022-01-02T10:00:00.000Z'),
+//     })
+//   })
+
+//   test('Skip produces correct file - no previous update', async () => {
+//     await handler({ skip: true })
+//     const updateData = JSON.parse(
+//       fs.readFileSync('/tmp/.redwood/update-data.json')
+//     )
+//     expect(updateData).toEqual(DEFAULT_UPDATE_DATA_JSON)
+//   })
+
+//   test('Skip produces correct file', async () => {
+//     await handler({})
+//     await handler({ skip: true })
+//     const updateData = JSON.parse(
+//       fs.readFileSync('/tmp/.redwood/update-data.json')
+//     )
+//     expect(updateData).toEqual({
+//       localVersion: '1.0.0',
+//       remoteVersion: '1.0.0',
+//       skipVersion: '1.0.0',
+//       upgradeAvailable: false,
+//       lastChecked: Date.parse('2022-01-02T12:00:00.000Z'),
+//       lastShown: Date.parse('2022-01-02T12:00:00.000Z'),
+//     })
+//   })
+
+//   test('Silent has no output', async () => {
+//     await handler({ silent: true })
+//     expect(console.log).not.toHaveBeenCalled()
+//   })
+
+//   test('No update message', async () => {
+//     await handler({})
+//     expect(
+//       console.log.mock.calls[console.log.mock.calls.length - 1][0]
+//     ).toEqual(expect.stringContaining('No upgrade is available'))
+//   })
+
+//   test('Creates correct update-data.json file', async () => {
+//     await handler({})
+//     const updateData = JSON.parse(
+//       fs.readFileSync('/tmp/.redwood/update-data.json')
+//     )
+//     expect(updateData).toEqual({
+//       localVersion: '1.0.0',
+//       remoteVersion: '1.0.0',
+//       skipVersion: '0.0.0',
+//       upgradeAvailable: false,
+//       lastChecked: Date.parse('2022-01-02T12:00:00.000Z'),
+//       lastShown: Date.parse('2022-01-02T12:00:00.000Z'),
+//     })
+//   })
+
+//   test('Lock is created', async () => {
+//     fs.writeFileSync = jest.fn()
+//     await handler({})
+//     expect(fs.writeFileSync.mock.calls[0][0]).toBe(
+//       '/tmp/.redwood/locks/update-command'
+//     )
+//   })
+
+//   test('Lock is removed', async () => {
+//     fs.unlinkSync = jest.fn()
+//     await handler({})
+//     expect(fs.unlinkSync.mock.calls[0][0]).toBe(
+//       '/tmp/.redwood/locks/update-command'
+//     )
+//   })
+// })
+
+describe('Upgrade is available (1.0.0 -> 2.0.0)', () => {
+  let consoleInfoSpy
+  beforeEach(() => {
+    jest.useFakeTimers()
+    jest.setSystemTime(Date.parse('2022-01-02T12:00:00.000Z'))
+    latestVersion.mockImplementation(() => {
+      return '2.0.0'
+    })
+    consoleInfoSpy = jest.spyOn(console, 'info').mockImplementation(() => {})
+    console.log = jest.fn()
```

Comment:
Doing it this way I don't think `mockRestore` will work. Jest won't know what `console.log` used to be.